libpromises/evalfunction.c

```diff
@@ -195,8 +196,9 @@ static VarRef* ResolveAndQualifyVarName(const FnCall *fp, const char *varname)
     return ref;
 }
 
-static JsonElement* VarRefValueToJson(EvalContext *ctx, const FnCall *fp, const VarRef *ref,
-                                      const DataType disallowed_datatypes[], size_t disallowed_count)
+static JsonElement* VarRefValueToJsonAllowingScalars(EvalContext *ctx, const FnCall *fp, const VarRef *ref,
+                                                     const DataType disallowed_datatypes[], size_t disallowed_count,
+                                                     bool allow_scalars)
```

Comment:
Why can't `disallowed_datatypes` be used here instead of introducing yet another argument `allow_scalars`? Can't you just put the scalar type as disallowed in the appropriate places?